pkg/model/pvcs.go

```diff
@@ -62,7 +65,31 @@ func (b *BackstagePvcs) validate(model *BackstageModel, backstage bsv1.Backstage
 
 func (b *BackstagePvcs) setMetaInfo(backstage bsv1.Backstage, scheme *runtime.Scheme) {
 	for _, item := range b.pvcs.Items {
-		item.(*corev1.PersistentVolumeClaim).Name = PvcsName(backstage.Name, item.(*corev1.PersistentVolumeClaim).Name)
-		setMetaInfo(item, backstage, scheme)
+		pvc := item.(*corev1.PersistentVolumeClaim)
+		pvc.Name = PvcsName(backstage.Name, pvc.Name)
+		setMetaInfo(pvc, backstage, scheme)
+	}
+}
+
+// implementation of BackstagePodContributor interface
+func (b *BackstagePvcs) updatePod(deployment *appsv1.Deployment) {
+
+	for _, pvc := range b.pvcs.Items {
+		volName := utils.ToRFC1123Label(pvc.GetName())
+		volSrc := corev1.VolumeSource{
+			PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+				ClaimName: pvc.GetName(),
+			},
+		}
+		deployment.Spec.Template.Spec.Volumes =
+			append(deployment.Spec.Template.Spec.Volumes, corev1.Volume{Name: volName, VolumeSource: volSrc})
+
+		c := &deployment.Spec.Template.Spec.Containers[0]
```

Comment:
It is a good question.
Actually I am planning to remake this logic, the issue is coming.
Put it this way (tempararily) since we have this (Containers[0]) almost everywhere, except new spec.deployment.
So, let's keep it for the time.